### PR TITLE
[v2] Fixed Empty Password Selection

### LINKED ISSUE #1594: [v2] Password section - Empty Password Works
In the Password section in V2 even with empty password it allows the clicking to next and with enter key,
It should be restricted

https://github.com/llaske/sugarizer/assets/116670999/8ca78835-94c4-4477-8a34-424c99725ded

### PR DESCRIPTION
**Pull request by Vishalk91-4**
Fixes #1594 

Fixed the Empty Selection of Password

https://github.com/llaske/sugarizer/assets/116670999/86442324-5037-4bba-a11c-de4626f7fbae

@llaske, could you please review it

### CODE REVIEW

File: v2/js/screens/loginscreen.js [hunk 1]
```diff
@@ -117,6 +117,7 @@ const LoginScreen = {
 
 	data() {
 		return {
+			min: 4,
```
Comment (llaske) [inline v2/js/screens/loginscreen.js:120]: `minPwdSize` should be a better name

File: v2/js/screens/loginscreen.js [hunk 2]
```diff
@@ -166,6 +167,12 @@ const LoginScreen = {
 		}
 	},
 
+	computed: {
+		minChoosePassword() {
+			return this.l10n.stringChoosePassword.replace('{{min}}', this.min);
+		},
+	},
+
```
Comment (llaske) [inline v2/js/screens/loginscreen.js:175]: Create a dedicated computed property is not the best way to handle localization here

File: v2/js/screens/loginscreen.js [hunk 3]
```diff
@@ -270,7 +277,10 @@ const LoginScreen = {
 					}
 				}
 				else if (this.index.currentIndex === 2) { // password
-					this.index.currentIndex++;
+					this.details.password = this.$refs.passwordInput.passwordText;
+					if (this.details.password.length >= 4) {
```
Comment (llaske) [inline v2/js/screens/loginscreen.js:281]: Why not using the constant created before?
What about the minimum server size mentioned here : https://github.com/llaske/sugarizer/issues/1594#issuecomment-2052471106 ?

**Review by llaske — CHANGES_REQUESTED** [review]
See my comments.

File: v2/js/screens/loginscreen.js [hunk 4]
```diff
@@ -28,8 +28,8 @@ const LoginScreen = {
 			<input ref="nameInput" type="text" name="name" class="input_field" v-model="details.name" @keyup="handleEnterKey">
 		</div>
 		<div id="loginscreen_password" class="column" v-show="index.currentIndex === 2">
-			<div class="firstscreen_text" id="pass_text">{{l10n.stringPassword}}</div>
-			<password ref="passwordInput" @passwordSet="handlePasswordSet"></password>
+		<div class="firstscreen_text" id="pass_text">{{minChoosePassword}}</div>
+		<password ref="passwordInput" @passwordSet="handlePasswordSet"></password>
```
[the remark this answers is not in the document]
Comment (Vishalk91-4) [reply v2/js/screens/loginscreen.js:32]: Tried with `get`, don't know why it was showing 
` ) missing indentation error`

Will check again on it

File: v2/js/screens/loginscreen.js (hunk 2 above)
Comment (Vishalk91-4) [reply v2/js/screens/loginscreen.js:175]: I will keep that in mind for the next time
Thanks for saying that

File: v2/js/screens/loginscreen.js (hunk 3 above)
Comment (Vishalk91-4) [reply v2/js/screens/loginscreen.js:281]: Coming to that, I also wanted to know like, if you could elaborate on what do you want to be done in min server size
I didn't get it completely

File: v2/js/screens/loginscreen.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (llaske) [reply v2/js/screens/loginscreen.js:32]: You can't do a function call in the template.
The value should be updated programmatically, see how it's done here: https://github.com/llaske/sugarizer/blob/feature/v2/v2/js/screens/loginscreen.js#L252

File: v2/js/screens/loginscreen.js [hunk 5]
```diff
@@ -28,8 +28,8 @@ const LoginScreen = {
 			<input ref="nameInput" type="text" name="name" class="input_field" v-model="details.name" @keyup="handleEnterKey">
 		</div>
 		<div id="loginscreen_password" class="column" v-show="index.currentIndex === 2">
-			<div class="firstscreen_text" id="pass_text">{{l10n.stringPassword}}</div>
-			<password ref="passwordInput" @passwordSet="handlePasswordSet"></password>
+		<div ref="pwdText" class="firstscreen_text" id="pass_text" v-html="l10n.stringPasswordLink"></div>
+		<password ref="passwordInput" @passwordSet="handlePasswordSet"></password>
```
Comment (llaske) [inline v2/js/screens/loginscreen.js:32]: Preserve indentation of the block

File: v2/js/screens/loginscreen.js [hunk 6]
```diff
@@ -270,7 +272,12 @@ const LoginScreen = {
 					}
 				}
 				else if (this.index.currentIndex === 2) { // password
-					this.index.currentIndex++;
+					this.details.password = this.$refs.passwordInput.passwordText;
+					const info = await sugarizer.modules.server.getServerInformation(this.details.serverAddress);
```
Comment (llaske) [inline v2/js/screens/loginscreen.js:276]: Could take time, it's better to use the previously loaded value.

**Review by llaske — CHANGES_REQUESTED** [review]
That's better.

**Comment by llaske** [comment]
Nice. Thanks.

**Pull request merged**